### PR TITLE
Introduce a configuration analyzer into Helm test which shows information about how discovery and scraping are performing

### PR DESCRIPTION
**Pull request by petewall**
Also, build a test image, rather than installing the dependencies over and over.

### CODE REVIEW

**Comment by petewall** [comment]
Haha. Now the CI is failing, not because the code is bad, but because the `apt-get update && apt-get install -y curl jq` slowed the tests long enough for the metrics to show up.

File: charts/k8s-monitoring/test/config-analysis.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+set -eo pipefail
+
+AGENT_HOST="${AGENT_HOST:-http://localhost:8080}"
+sources=$1
+
+function discoveryRelabel() {
+    component=$1
```
Comment (skl) [inline charts/k8s-monitoring/test/config-analysis.sh:9]: May not be necessary but I think there might be some scope leakage here, so you could use `local`?

```suggestion
    local component=$1
```

File: charts/k8s-monitoring/test/config-analysis.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+set -eo pipefail
+
+AGENT_HOST="${AGENT_HOST:-http://localhost:8080}"
+sources=$1
```
Comment (skl) [inline charts/k8s-monitoring/test/config-analysis.sh:6]: Maybe a `test -z` or similar if this is a required arg?

**Review by skl — APPROVED** [review]
💯

File: charts/k8s-monitoring/test/config-analysis.sh (hunk 1 above)
Comment (petewall) [reply charts/k8s-monitoring/test/config-analysis.sh:9]: Good call.

**Pull request merged**